.evergreen/vs-env-run.ps1

```diff
@@ -0,0 +1,246 @@
+    #         }
+    #         return & $vswhere_found -utf8 -nologo -format json -all -legacy -prerelease -products * `
+    #         | ConvertFrom-Json `
+    #         | ForEach-Object { $_.installationVersion } `
+    #         | Where-Object { $_ -like "$wordToComplete*" }
+    #     })]
+    [string]
+    $Version,
+    # The target architecture for the build
+    [Parameter(Mandatory)]
+    [ValidateSet("x86", "amd64", "arm", "arm64", IgnoreCase = $false)]
+    [string]
+    $TargetArch,
+    # Select a specific Windows SDK version.
+    [string]
+    # xxx: This requires PowerShell 5+, which some build hosts don't have:
+    # [ArgumentCompleter({
+    #         param($commandName, $paramName, $wordToComplete, $commandAst, $fakeBoundParameters)
+    #         $found = @()
+    #         if (Test-Path "${env:ProgramFiles(x86)}\Windows Kits\10\Include") {
+    #             $found += $(
+    #                 Get-ChildItem "${env:ProgramFiles(x86)}\Windows Kits\10\Include"`
+    #                 | Where-Object { Test-Path "$($_.FullName)\um\Windows.h" }`
+    #                 | ForEach-Object { Split-Path -Leaf $_.FullName }
+    #             )
+    #         }
+    #         if (Test-Path "${env:ProgramFiles(x86)}\Windows Kits\8.1\Include") {
+    #             $found += $(
+    #                 Get-ChildItem "${env:ProgramFiles(x86)}\Windows Kits\8.1\Include"`
+    #                 | Where-Object { Test-Path "$($_.FullName)\um\Windows.h" }`
+    #                 | ForEach-Object { Split-Path -Leaf $_.FullName }
+    #             )
+    #         }
+    #         return $found | Where-Object { $_ -like "$wordToComplete*" }
+    #     })]
+    $WinSDKVersion,
+    # The host architecture to use. Not usually needed. Defaults to x86.
+    [ValidateSet("x86", "amd64", IgnoreCase = $false)]
+    [string]
+    $HostArch = "x86",
+    # The Visual C++ toolset to load
+    [string]
+    $VCToolsetVersion,
+    # Prefer Visual C++ libraries with Spectre mitigations
+    [switch]
+    $UseSpectreMiticationLibraries,
+    # The app platform to load. Default is "Desktop"
+    [ValidateSet("Desktop", "UWP", IgnoreCase = $false)]
+    [string]
+    $AppPlatform = "Desktop",
+    # The directory to store ephemeral files
+    [string]
+    $ScratchDir,
+    # The command to execute within the VS environment. May be any invocable object.
+    [Parameter(Mandatory, Position = 1)]
+    $Command
+)
+
+$ErrorActionPreference = 'Stop'
+
+$this_dir = $PSScriptRoot
+
+if ([string]::IsNullOrEmpty($ScratchDir)) {
+    $ScratchDir = Join-Path (Split-Path -Parent $this_dir) "_build"
+}
+
+New-Item $ScratchDir -ItemType Directory -ErrorAction Ignore
+$vswhere = Join-Path $ScratchDir "vswhere.exe"
+
+$ProgressPreference = "SilentlyContinue"
+[Net.ServicePointManager]::SecurityProtocol = 'tls12, tls11'
+Invoke-WebRequest `
+    -UseBasicParsing `
+    -Uri "https://github.com/microsoft/vswhere/releases/download/3.0.3/vswhere.exe" `
+    -OutFile $vswhere
```

Comment:
Should this check if `$vswhere` exists first? What if `vs-env-run.ps1` is invoked twice with the same scratch directory?

Reply:
It's fine if the file already exists, but a tiny perf boost to not re-download it